kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -254,6 +274,44 @@ private AtomicWriteResult flush(final long offset, final UUID txnid) {
     );
     buffer.clear();
 
+    maybeTruncateChangelog(offset);
+    return AtomicWriteResult.success(partition);
+  }
+
+  private AtomicWriteResult drain(
+      final Iterator<CompletionStage<AtomicWriteResult>> it
+  ) {
+    try {
+      final var inFlight = new ArrayList<CompletableFuture<AtomicWriteResult>>();
+
+      while (it.hasNext()) {
+        inFlight.add(it.next().toCompletableFuture());
+        if (inFlight.size() >= ResponsiveConfig.MAX_CONCURRENT_REMOTE_WRITES_DEFAULT) {
```

Comment:
we never clear `inFlight` - so I think after the first batch this will just send one request at a time.

Reply:
oops! I'll add a test for this and extract it into a utility - make sure the above is tested as well.

Reply:
no i don't think so, my original testing I implemented this correctly - I refactored it to make it cleaner and lost some in translation :( 

This was the testing code, you can see I used `% MAX_CONCURRENT_WRITES` instead of checking size:

```
   final var flushes = new ArrayList<CompletableFuture<FlushResult>>();
     for (PartitionedBuilder builder : builders.values()) {
       var stage = builder.flush();
       var result = stage.thenCompose(
           fr -> !fr.wasApplied
               ? CompletableFuture.completedStage(fr)
               : finalize(fr, txnid, offset));
       flushes.add(result.toCompletableFuture());

       if (flushes.size() % MAX_CONCURRENT_WRITES == 0) {
         for (CompletableFuture<FlushResult> flush : flushes) {
           try {
             flush.get();
           } catch (InterruptedException | ExecutionException e) {
             throw new RuntimeException(e);
           }
         }
       }
     }
```